python/graphstorm/model/utils.py

```diff
@@ -1686,3 +1688,45 @@ def append_to_dict(from_dict, to_dict):
             to_dict[k].append(v.cpu())
         else:
             to_dict[k] = [v.cpu()]
+
+def normalize_node_embs(embs, norm_method, is_train=False):
+    """ Do node embedding normalization
+
+        Parameters
+        ----------
+        embs: dict of Tensor
+            node embeddings.
+        norm_method: str
+            Node embedding normalization method.
+        is_train: bool
+            Whether it is called during training.
+    """
+    if norm_method is None or norm_method == "":
+        def norm(emb):
+            return emb
+        norm_func = norm
+    elif norm_method == GRAPHSTORM_LP_EMB_L2_NORMALIZATION:
+        def l2_norm(emb):
+            if is_train is True:
+                # When it is training, we expect the size of emb is small
+                return F.normalize(emb)
+
+            rank = get_rank()
+            world_size = get_world_size()
+            if isinstance(emb, (dgl.distributed.DistTensor, LazyDistTensor)):
+                start, end = get_data_range(rank, world_size, len(emb))
+            else:
+                start, end = 0, len(emb)
```

Comment:
when this happens? if emb is a normal tensor, why not call F.normalize directly?

Reply:
In minibatch inference, it will be a normal tensor.
The tensor may be super large, call F.normalize may cause some problems.